Entities/Common/Controls/ImportantPickup.as

```diff
@@ -0,0 +1,102 @@
+#include "StandardControlsCommon.as"
+
+// Current baseZ ordening in the game:
+// 100: door OPEN, bridge OPEN
+// -50+: buildings
+// -40 ladders
+// relative -10: pickup behind (keg, crate)
+// -25 ballista
+// -20: fireplace
+// -10: saw (-1,-5)
+//      chest
+//      mounted bow
+// 10: carrying saw
+//     bison
+//     shark
+//     bush
+//     grain, flower
+// 20: DEFAULT IMPORTANT
+// 30: bomb arrows
+// 50: workbench foreground (weird)
+// 100: door CLOSED, bridge CLOSED
+// 500: spike
+
+
+// Add the following to the Init of the object to specify a custom hover polygon:
+// Vec2f[] hoverPoly =
+//   { Vec2f(-7.0f, 3.5f)
+//   , Vec2f(-2.5f, -5.0f)
+//   , Vec2f(1.0f, -5.0f)
+//   , Vec2f(6.5f,  3.5f)
+//   , Vec2f(-7.0f, 3.5f)
+//   };
+
+// this.set("hover-poly", hoverPoly);
+
+
+// Overwrite importance by setting "important-pickup" in the onInit
+
+void onInit(CBlob@ this)
+{
+  f32 baseZ = 20.0f;
+  if (this.exists("important-pickup")) 
+  {
+    baseZ = this.get_f32("important-pickup");
+  }
+
+  this.getSprite().SetZ(baseZ);
+  this.set_f32("important-pickup", baseZ);
+}
+
+void onAttach(CBlob@ this, CBlob@ attached, AttachmentPoint @attachedPoint)
+{
+  f32 baseZ = this.getSprite().getZ();
+  this.set_f32("important-pickup", baseZ);
```

Comment:
Why do we need to check the sprite Z on attach? Looking at the logic my assumption would be that the sprite Z is not expected to change, so that `"important-pickup"` should always be correct.

This is not *really* important but is this something you've seen happen?

Reply:
I removed `onAttach` for now. It does not make sense at the moment to modify the baseZ. Maybe just old code